src/Hamiltonians/HubbardReal1D.jl

```diff
@@ -94,6 +94,38 @@ function diagonal_element(h::HubbardReal1D, address::BoseFS)
 end
 
 function get_offdiagonal(h::HubbardReal1D, add::BoseFS, chosen)
-    naddress, onproduct = hopnextneighbour(add, chosen)
-    return naddress, - h.t * sqrt(onproduct)
+    new_address, on_product = hopnextneighbour(add, chosen)
+    return new_address, -h.t * √on_product
 end
+
+###
+### 2D Model
+###
+function num_offdiagonals(::HubbardReal1D, address::BoseFS2D)
+    return 4 * numberoccupiedsites(address.bosefs)
+end
```

Comment:
Hmm, so it seems like setting one of the dimensions of the 2D lattice to one does not give you back the same result like a 1D model. Should this be so?

Reply:
The next case to think about is if one of the dimensions is 2. The current code would calculate hop along this direction twice.